cms/templates/asset_index.html

```diff
@@ -42,15 +43,28 @@ <h1 class="page-header">
 
         <nav class="nav-actions" aria-label="${_('Page Actions')}">
             <h3 class="sr">${_("Page Actions")}</h3>
-            <ul>
-                <li class="nav-item">
-                    <a href="#" class="button upload-button new-button"><span class="icon fa fa-plus" aria-hidden="true"></span> ${_("Upload New File")}</a>
-                </li>
-            </ul>
+
+            <form class="filter-criteria-form" action="${asset_callback_url}" method="post">
+                <input type="hidden" name="csrfmiddlewaretoken" value="${csrf_token}"/>
+                <ul>
+                    <li class="nav-item">
+                        <input type="text" name="filter_criteria" class="filter-criteria-input filter-criteria-clearable" value="${filter_criteria}"/>
+                    </li>
+                    <li class="nav-item">
+                        <button type="submit" class="button new-button filter-criteria-button"><span aria-hidden="true"></span> Filter</button>
```

Comment:
The Submit button is going to need to be associated with the input. Take a look at search examples on this page (you can see the HTML on the tab). http://ux.edx.org/components/forms/